test/e2e/util/external_oidc.go

```diff
@@ -0,0 +1,334 @@
+package util
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/gob"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1typedclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+
+	kauthnv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kauthnv1typedclient "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	"k8s.io/client-go/rest"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ProviderType string
+
+const (
+	ProviderAzure    ProviderType = "azure"
+	ProviderKeycloak ProviderType = "keycloak"
+)
+
+type ExtOIDCConfig struct {
+	ExternalOIDCProvider     ProviderType
+	OIDCProviderName         string
+	CliClientID              string
+	ConsoleClientID          string
+	IssuerURL                string
+	GroupPrefix              string
+	UserPrefix               string
+	ConsoleClientSecretName  string
+	ConsoleClientSecretValue string
+
+	//format: “user1:psw1,user2:psw2”, it is used for keycloak oidc
+	TestUsers string
+
+	//for oidcProviders.issuer.issuerCertificateAuthority
```

Comment:
@heliubj18 , continuing https://github.com/openshift/hypershift/pull/6419#discussion_r2253053281 `many occurrences of "//xxx"`, how about checking others and updating too?
test/e2e/util/external_oidc.go:
```
//format: “user1:...
//for oidcProviders...

test/e2e/util/options.go
//external oidc for ...
//set external OIDC if enabled

test/e2e/util/hypershift_framework.go
//create external oidc secret and configmap
```